ext/reflection/tests/readonly_properties.phpt

```diff
@@ -0,0 +1,24 @@
+--TEST--
+Readonly property reflection
+--FILE--
+<?php
+
+class Test {
+    public int $rw;
+    public readonly int $ro;
+}
+
+$rp = new ReflectionProperty(Test::class, 'rw');
+var_dump($rp->isReadOnly());
+var_dump(($rp->getModifiers() & ReflectionProperty::IS_READONLY) != 0);
+
+$rp = new ReflectionProperty(Test::class, 'ro');
+var_dump($rp->isReadOnly());
+var_dump(($rp->getModifiers() & ReflectionProperty::IS_READONLY) != 0);
+
```

Comment:
`setValue` by reflection should be tested here and allowed

Reply:
>  ReflectionProperty::setValue() can bypass the requirement that initialization occurs from the scope where the property has been declared. However, reflection cannot modify a readonly property that has already been initialized. 

Allowing `setValue` on properties that have already been set breaks the assumption that the value can never change. This assumption is useful for not only the user but also the optimizer.

Reply:
But php is known for it's beaty that reflection non-constants can be changed during runtime which makes debugging and hacking much easier.

Reply:
Nikita's standpoint has always been that we shouldn't allow such loopholes, and I also agree with him - Ilya pointed out the reasoning very clearly. PHP used to allow a lot of shady things in the past, but it's less and less true these days.